lisa/tests/scheduler/load_tracking.py

```diff
@@ -738,11 +739,24 @@ def _plot_behaviour(self, ax, df, title, task_duty_cycle_pct):
         ax.legend()
 
     def _test_behaviour(self, signal_name, error_margin_pct, allowed_error_pct):
+	# Note: This test-case is only valid if executed at capacity == 1024
```

Comment:
Can we assert for this condition here? It seems we have a list of cpu capacities in self.plat_info['cpu-capacities'], and we read the cpu values below - so we can check that the capacity of that cpu == 1024?

Reply:
IIUC, plat_info just tells us if which capacities that are available. The PELT test-cases require that the actual capacity is 1024 throughout the trace. To create an assertion for that we need to trace the capacity. We can do that as separate value-add if needed. I just put the note there to remind people of a reason why this test-case could break. I should have added it to _test_range() too as it applied to that as well.

Reply:
That should be fairly easy to do, the question is whether we want to raise an exception or return a failed test (@douglas-raillard-arm ?).

Since we get the CPU associated with the task a few lines below, we could do this:
```python
cpu = phase.cpus[0]

if self.plat_info["cpu-capacities"][cpu] != UTIL_SCALE:
    # raise RuntimeError or return a ResultBundle.from_bool(False) or something
```

Reply:
But as said above, that only checks the max capacity of the CPU. We need to verify that the CPU did indeed run at that capacity throughout the test.

Reply:
Right, I get it now. Since we'd need extra trace events, that's fine as is.

Reply:
It is fine to add the check to catch at least some cases.

Reply:
Done for both range tests and behaviour tests.